crates/goose-cli/src/cli.rs

```diff
@@ -59,6 +59,14 @@ fn extract_identifier(identifier: Identifier) -> session::Identifier {
     }
 }
 
+fn parse_key_val(s: &str) -> Result<(String, String), String> {
+    let parts: Vec<&str> = s.splitn(2, '=').collect();
+    if parts.len() != 2 {
+        return Err(format!("invalid KEY=VALUE: {}", s));
+    }
+    Ok((parts[0].to_string(), parts[1].to_string()))
+}
```

Comment:
could simplify with `split_once` here
```suggestion
    match s.split_once('=') {
        Some((k, v)) => Ok((k.to_string(), v.to_string())),
        None => Err(format!("invalid KEY=VALUE: no `=` in `{}`", s)),
    }
```